src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -337,6 +338,18 @@ trait PodsValidation {
     pod.scaling is optional(scalingValidator)
     pod is endpointNamesUnique and endpointContainerPortsUnique and endpointHostPortsUnique
   }
+
+  // Marathon 1.4.0 did not introduce a validation rule for container names.
+  // Introducing this rule needs to allow existing (possibly wrong) container names.
+  // A migration can not fix this problem, since the task id is generated from the container name
+  // and there might be started tasks.
+  // In a later version this validation rule should be applied in the default validator.
+  val validPodContainerUpdate: Validator[MesosContainer] = validator[MesosContainer] { container =>
+    container.name should matchRegexFully("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  }
+  val validPodUpdate: Validator[PodDefinition] = validator[PodDefinition] { pod =>
```

Comment:
Marathon 1.5 validates container names (length and content) during RAML parsing: should this be considered a breaking change?

If we're adding container name validation for 1.4, why not just add it here instead? https://github.com/mesosphere/marathon/blob/d16b4547c118769c4ec6054c286acac61dd101e6/src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala#L189